lib/Sema/PreCheckExpr.cpp

```diff
@@ -1066,11 +1068,27 @@ namespace {
       // Resolve 'super' references.
       if (auto *superRef = dyn_cast<SuperRefExpr>(expr)) {
         auto loc = superRef->getLoc();
+
         auto *selfDecl = getImplicitSelfDeclForSuperContext(loc);
         if (selfDecl == nullptr)
           return finish(true, new (Ctx) ErrorExpr(loc));
 
         superRef->setSelf(selfDecl);
+
+        // Diagnose a 'super' reference that is not the base of a subscript or
+        // dot expression.
+        // NB: This is done along the happy path because presenting this error
+        // in a context where 'super' is not legal to begin with is not helpful.
+        auto *parentExpr = Parent.getAsExpr();
+        if (!(parentExpr && (isa<SubscriptExpr>(parentExpr) ||
```

Comment:
I think you may also have to ensure that it's the base of the subscript, e.g `x[super]` should be invalid

Reply:
Oh, good point. Will add a test case too.